src/cli/Cli.ts

```diff
@@ -974,7 +974,7 @@ export class Cli {
   public static async handleMultipleResultsFound(promptMessage: string, errorMessage: string, values: { [key: string]: object }): Promise<object | CommandError> {
     const prompt: boolean = Cli.getInstance().getSettingWithDefaultValue<boolean>(settingsNames.prompt, false);
     if (!prompt) {
-      throw errorMessage;
+      throw new Error(errorMessage);
```

Comment:
My proposition is to change this function slightly. For one it does not return a CommandError, (it only throws an error) and secondly we need annoying `as`-typings in the files that use this function.

What do you think about using generics here?:

```typescript
public static async handleMultipleResultsFound<T>(promptMessage: string, errorMessage: string, values: { [key: string]: T }): Promise<T> {

  // function code
  
}
```

We can then use it as follows (example from app-get.ts):

```typescript
const result = await Cli.handleMultipleResultsFound<{ id: string }>(`Multiple Azure AD application registration with name '${name}' found. Choose the correct ID:`, `Multiple Azure AD application registration with name '${name}' found.`, resultAsKeyValuePair);
  return result.id;
```

Reply:
sure. good idea. I will try to process this this week.